### PR TITLE
Automated the dowloading of ReadFromDocs procedure and Dockerized the current backend.

### PR DESCRIPTION
**Pull request by SarveshAtawane**
Key changes:
- Added the file fetchAndOrganizeData.py. This file collects data from all the links provided in config.yaml. The collected data is then organized in a dedicated folder containing all the HTML files gathered.
- In ingest.py, we've added some debugging code to print the total number of chunks. Additionally, the creation of the chromadb folder is now automated.
- I've also included a Dockerfile and added usage commands for both Docker and non-Docker environments.

@gcapuzzi @swaptr Can you please review this?

### CODE REVIEW

**Comment by swaptr** [comment]
Thank you for submitting the PR, it looks very promising. However, this PR is failing the DCO check. Please sign your commits to resolve this issue.

File: src/core/ingest.py [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
-# save data into a directory
-vectorstore = Chroma.from_documents(documents=splits, embedding=embeddings, persist_directory=config_data["persist_directory"])
-vectorstore.persist()
\ No newline at end of file
+
+# Print the number of chunks
+print(f"Number of chunks created: {len(splits)}")
+
+# Add a delay
+time.sleep(5)  # 2-second delay
```
Comment (swaptr) [inline src/core/ingest.py:34]: Why do we need to add a delay here?

File: src/core/fetchAndOrganizeData.py
Comment (swaptr) [inline src/core/fetchAndOrganizeData.py:1]: Could you kindly limit the scope of this pull request to just dockerization? For changes related to data ingestion and automation, please submit a separate pull request. While data automation is a priority, this PR cannot proceed as model development is not finished yet. Thank you.

**Review by swaptr — CHANGES_REQUESTED** [review]

File: src/core/fetchAndOrganizeData.py
[the remark this answers is not in the document]
Comment (SarveshAtawane) [reply src/core/fetchAndOrganizeData.py:1]: Ok Sure.

File: src/core/requirements.txt [hunk 2]
```diff
@@ -5,4 +5,7 @@ sentence-transformers
 chromadb
 bitsandbytes
 accelerate
-uvicorn
\ No newline at end of file
+uvicorn
+beautifulsoup4
```
[the remark this answers is not in the document]
Comment (SarveshAtawane) [reply src/core/requirements.txt:9]: Uvicorn was previously there it just got retyped and therefore it happens to like it is added. It is for the FastAPI server.
Oh I missed the bs4 I was trying other things and forgot to remove it.

File: src/core/ingest.py (hunk 1 above)
Comment (SarveshAtawane) [reply src/core/ingest.py:34]: It was just for understaning how much chunks have been created. I will remove it.

File: src/core/fetchAndOrganizeData.py
Comment (SarveshAtawane) [reply src/core/fetchAndOrganizeData.py:1]: @swaptr I have made the changes can you check again. Thankyou.

Comment (swaptr) [reply src/core/fetchAndOrganizeData.py:1]: Looks great @SarveshAtawane!

File: src/core/ingest.py [hunk 3]
```diff
@@ -1,29 +1,35 @@
+import os
 import torch
 from utils import load_yaml_file
 from langchain_community.document_loaders.recursive_url_loader import RecursiveUrlLoader
 from langchain_community.document_loaders import ReadTheDocsLoader
 from langchain_community.document_loaders.merge import MergedDataLoader
 from langchain.text_splitter import RecursiveCharacterTextSplitter
 from langchain.vectorstores import Chroma
+import time
```
Comment (swaptr) [inline src/core/ingest.py:9]:
```suggestion

```
This can be safely removed.

Comment (SarveshAtawane) [reply src/core/ingest.py]: Ohh sorry. I changed it.

**Comment by swaptr** [comment]
LGTM. Thanks for your contribution. :rocket:
